src/main/java/com/commercetools/sync/customers/utils/CustomerUpdateActionUtils.java

```diff
@@ -0,0 +1,994 @@
+     */
+    @Nonnull
+    private static List<UpdateAction<Customer>> buildRemoveStoreUpdateActions(
+        @Nonnull final List<KeyReference<Store>> oldStores,
+        @Nonnull final List<ResourceIdentifier<Store>> newStores) {
+
+        final Map<String, ResourceIdentifier<Store>> newStoreKeyToStoreMap =
+            newStores.stream()
+                     .filter(Objects::nonNull)
+                     .filter(storeResourceIdentifier -> storeResourceIdentifier.getKey() != null)
+                     .collect(toMap(ResourceIdentifier::getKey, identity()));
+
+        return oldStores
+            .stream()
+            .filter(Objects::nonNull)
+            .filter(storeKeyReference -> !newStoreKeyToStoreMap.containsKey(storeKeyReference.getKey()))
+            .map(storeKeyReference -> RemoveStore.of(ResourceIdentifier.ofKey(storeKeyReference.getKey())))
+            .collect(toList());
+    }
+
+    /**
+     * Compares the {@link List} of {@link Store} {@link KeyReference}s and {@link Store} {@link ResourceIdentifier}s
+     * of a {@link CustomerDraft} and a {@link Customer}. It returns a {@link List} of {@link AddStore} update actions
+     * as a result, if the old store needs to be added to a customer to have the same set of stores as the new customer.
+     * If both the {@link Customer} and the {@link CustomerDraft} have the same set of stores, then no update actions
+     * are needed and hence an empty {@link List} is returned.
+     *
+     * <p>Note: Null values of the stores are filtered out.
+     *
+     * @param oldStores the stores which should be updated.
+     * @param newStores the stores where we get the new store.
+     * @return A list containing the update actions or an empty list if the store references are identical.
+     */
+    @Nonnull
+    private static List<UpdateAction<Customer>> buildAddStoreUpdateActions(
+        @Nonnull final List<KeyReference<Store>> oldStores,
+        @Nonnull final List<ResourceIdentifier<Store>> newStores) {
+
+        final Map<String, KeyReference<Store>> oldStoreKeyToStoreMap =
+            oldStores.stream()
+                     .filter(Objects::nonNull)
+                     .collect(toMap(KeyReference::getKey, identity()));
+
+        return newStores
+            .stream()
+            .filter(Objects::nonNull)
+            .filter(storeResourceIdentifier -> !oldStoreKeyToStoreMap.containsKey(storeResourceIdentifier.getKey()))
+            .map(AddStore::of)
+            .collect(toList());
+    }
+
+    /**
+     * Compares the stores of a {@link Customer} and a {@link CustomerDraft}. It returns a {@link List} of
+     * {@link UpdateAction}&lt;{@link Customer}&gt; as a result. If no update action is needed, for example in
+     * case where both the {@link Customer} and the {@link CustomerDraft} have the identical stores, an empty
+     * {@link List} is returned.
```

Comment:
The method is about address update action. Here the javadoc is not describing the same thing.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/591/commits/20f7acf6e64dd5b8b5a7727b4ca0f28a0ea76b76